xarray/core/accessor_str.py

```diff
@@ -846,68 +1085,881 @@ def replace(self, pat, repl, n=-1, case=None, flags=0, regex=True):
+        If the separator is not found, return 3 elements containing two empty strings,
+        followed by the string itself.
+
+        This is equivalent to :meth:`str.rpartion`.
+
+        Parameters
+        ----------
+        dim : Hashable or `None`
+            Name for the dimension to place the 3 elements in.
+            If `None`, place the results as list elements in an object DataArray
+        sep : str, default `" "`
+            String to split on.
+
+        Returns
+        -------
+        rpartitioned : same type as values or object array
+
+        See Also
+        --------
+        DataArray.str.partition
+        str.rpartition
+        pandas.Series.str.rpartition
+        """
+        return self._partitioner(func=self._obj.dtype.type.rpartition, dim=dim, sep=sep)
+
+    def _splitter(
+        self,
+        func: Callable,
+        pre: bool,
+        dim: Hashable,
+        sep: Optional[Union[str, bytes]],
+        maxsplit: int,
+    ) -> Any:
+        """
+        Implements logic for `split` and `rsplit`.
+        """
+        if sep is not None:
+            sep = self._stringify(sep)
+
+        if dim is None:
+            f = lambda x: func(x, sep, maxsplit)
+            return self._apply(f, dtype=np.object_)
+
+        # _apply breaks on an empty array in this case
+        if not self._obj.size:
+            return self._obj.copy().expand_dims({dim: 0}, -1)
+
+        f_count = lambda x: max(len(func(x, sep, maxsplit)), 1)
+        maxsplit = self._apply(f_count, dtype=np.int_).max().data.tolist() - 1
+
+        def _dosplit(mystr, sep=sep, maxsplit=maxsplit, dtype=self._obj.dtype):
+            res = func(mystr, sep, maxsplit)
+            if len(res) < maxsplit + 1:
+                pad = [""] * (maxsplit + 1 - len(res))
+                if pre:
+                    res += pad
+                else:
+                    res = pad + res
+            return np.array(res, dtype=dtype)
+
+        # dtype MUST be object or strings can be truncated
+        # See: https://github.com/numpy/numpy/issues/8352
+        return self._apply(
+            _dosplit,
+            dtype=np.object_,
+            output_core_dims=[[dim]],
+            output_sizes={dim: maxsplit},
+        ).astype(self._obj.dtype.kind)
+
+    def split(
+        self,
+        dim: Optional[Hashable],
+        sep: Union[str, bytes] = None,
+        maxsplit: int = -1,
+    ) -> Any:
+        """
+        Split strings in a DataArray around the given separator/delimiter `sep`.
+
+        Splits the string in the DataArray from the beginning,
+        at the specified delimiter string.
+
+        This is equivalent to :meth:`str.split`.
+
+        Parameters
+        ----------
+        dim : Hashable or `None`
+            Name for the dimension to place the results in.
+            If `None`, place the results as list elements in an object DataArray
+        sep : str, default is split on any whitespace.
+            String to split on.
```

Comment:
This would be interpreted as a (broken) link to `is split on any whitespace` by `napoleon`, so I would move it to the parameter description:
```suggestion
        sep : str, default: None
            String to split on. If ``None``, split on any whitespace.
```

Reply:
Fixed